backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,259 @@
+
+        if (application is null)
+        {
+            return new UpdateApplicationPayload(
+                new UpdateApplicationError(
+                    UpdateApplicationErrorCode.UNKNOWN_APPLICATION,
+                    "Unknown application.",
+                    new[] { nameof(input), nameof(input.ApplicationId).FirstCharToLower() }
+                )
+            );
+        }
+
+        var descriptor = new OpenIddictApplicationDescriptor();
+        await applicationManager.PopulateAsync(descriptor, application, cancellationToken).ConfigureAwait(false);
+        UpdateApplicationDescriptor(input, descriptor);
+        await applicationManager.UpdateAsync(application, descriptor, cancellationToken).ConfigureAwait(false);
+
+        return new UpdateApplicationPayload(application);
+    }
+
+    [UseUserManager]
+    [Authorize(Policy = AuthConfiguration.WritePolicy)]
+    public async Task<DeleteApplicationPayload> DeleteApplicationAsync(
+        DeleteApplicationInput input,
+        OpenIddictApplicationManager<OpenIdApplication> applicationManager,
+        ClaimsPrincipal claimsPrincipal,
+        UserManager<User> userManager,
+        ApplicationDbContext context,
+        CancellationToken cancellationToken
+    )
+    {
+        if (!await OpenIdConnectAuthorization.IsAuthorizedToManageApplication(
+                input.ApplicationId,
+                applicationManager,
+                claimsPrincipal,
+                userManager,
+                context,
+                cancellationToken
+            ).ConfigureAwait(false))
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(
+                    DeleteApplicationErrorCode.UNAUTHORIZED,
+                    "You are not authorized to delete the application.",
+                    new[] { nameof(input), nameof(input.ApplicationId).FirstCharToLower() }
+                )
+            );
+        }
+        if (input.ApplicationId == Guid.Empty)
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(DeleteApplicationErrorCode.UNKNOWN,
+                "Empty Application Id",
+                Array.Empty<string>()));
+        }
+
+        var application = await applicationManager.FindByIdAsync(input.ApplicationId.ToString(), cancellationToken).ConfigureAwait(false);
+
+        if (application is null)
+        {
+            return new DeleteApplicationPayload(
+                new DeleteApplicationError(
+                    DeleteApplicationErrorCode.UNKNOWN_APPLICATION,
+                    "Unknown application.",
+                    new[] { nameof(input), nameof(input.ApplicationId).FirstCharToLower() }
+                )
+            );
+        }
+
+        await applicationManager.DeleteAsync(application, cancellationToken).ConfigureAwait(false);
+
+        DeleteApplicationInstitutionConnection(application, context, cancellationToken);
```

Comment:
In `ApplicationDbContext` we use cascading for this association. This should take care of deleting associations of the deleted application. Is this manual deletion process still necessary?